docs/reference/mapping/types/parent-join.asciidoc

```diff
@@ -0,0 +1,363 @@
+[[mapping-parent-join-field]]
+=== `join` datatype
+
+The `join` datatype is a special field that creates
+parent/child relation within documents of the same index.
+This field defines a set of possible relations within the documents,
+each relation being a parent name and a child name.
+A single parent/child relation can be defined as follow:
+
+[source,js]
+--------------------------------------------------
+PUT my_index
+{
+  "mappings": {
+    "doc": {
+      "properties": {
+        "my_join_field": { <1>
+          "type": "join",
+          "my_parent": "my_child" <2>
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+
+<1> The name for the field
+<2> `my_parent` is parent of `my_child`.
```

Comment:
Maybe it'd be more clear to index a parent with some children. Then it'd be more obvious how `my_parent` and `my_child` are used. It looks like `my_parent` as a key has special meaning and I'm not used to that sort of thing in mappings. We do it all the time in queries though.